crates/polars-plan/src/logical_plan/optimizer/predicate_pushdown/join.rs

```diff
@@ -133,12 +133,19 @@ pub(super) fn process_join(
             // join columns, we also push down right
             || filter_left
                 && all_pred_cols_in_left_on(predicate, expr_arena, &left_on)
-                // TODO: Restricting to Inner and Left Join is probably too conservative
                 && matches!(&options.args.how, JoinType::Inner | JoinType::Left)
         {
             insert_and_combine_predicate(&mut pushdown_right, predicate, expr_arena);
             filter_right = true;
         }
+        #[cfg(feature = "semi_anti_join")]
+        if filter_left
+            && all_pred_cols_in_left_on(predicate, expr_arena, &left_on)
+            && matches!(&options.args.how, JoinType::Semi)
+        {
+            insert_and_combine_predicate(&mut pushdown_right, predicate, expr_arena);
```

Comment:
Now we do an extra traversal. Can you rewrite the `matches!` on line 136 to a proper `match` statement where we put the Semi as well?

Reply:
That's what I did in my initial commit but I didn't get where to put the feature gate. Can I put it inside the if statement ?

Reply:
Something like this:

```rust
match how {
  #[cfg(feature = ..)]
  JoinType::Semi,
  JoinType::Left
  ...
}
```